### ISSUE TITLE
Notes not always saved when using Dropbox

### ISSUE THREAD

**Issue by TheEnigmaBlade** [notable/notable#1163]
- **OS Version**: Windows 10 1909
- **Notable Version**: 1.8.4

### Current behavior

Files are randomly not saved fully when saving to Dropbox.

When this issue occurs, a new file is generated with a numeric suffix appended to the file name. This new file contains the latest changes to the note, while the original note file does not. Compared to other issues I've found, there is no complete data loss as the changes remain in the temporary file.

![](https://i.imgur.com/gQ5sxrZ.png "Example of files saved")

I assume this new file is a temporary file used to prevent data loss, however it is not replacing the original file. I've observed this working as expected when the data directory is moved out of Dropbox, so I'm guessing the issue is a race condition or file locking specific to Dropbox/other file sync services.

**Comment by fabiospampinato** [comment]
> I assume this new file is a temporary file used to prevent data loss

Yeah, the app is writing files "atomically" (i.e. in an indivisible action, it will either succeed or fail, avoiding partial file writes entirely), those files with numbers at the end are the temporary files the app writes containing the new content, after that's done those new files are renamed as the actual files basically.

I'm not sure how this fails or how we can fix it 🤔

**Comment by geoffreydv** [comment]
I'm having the same issue. It has been a while since I used notable so I can't really help narrow down the issue datewise. 

I have always been saving to a folder managed by dropbox, also with version 1.8.4 and that used to work. My guess is that either dropbox or windows have pushed an update that broke saving the files.

As the app is unusable now as all changes to the main .md files are lost I'm switching over to another folder on my hard drive which seems to work just fine.

**Comment by fabiospampinato** [comment]
It sounds like a serious issue. I actually use Dropbox myself too, but I'm on macOS, and I haven't seen any of these problems yet 🤔 There's probably something we can do about that on our hand, let's first see if once v1.9 gets released that addresses this already.

**Comment by stieben** [comment]
I have similar problems, also with Dropbox. Sometimes, it creates several temporary files. Sometimes, the most recent state is actually in the main file. And sometimes, but rarely, there's actual data loss or an inconsistency between Notable and the file contents, as if Notable is caching the contents somewhere.

Synchronising Dropbox manually seems to solve these issues, and that's what I started doing for now.
